wal.go

```diff
@@ -429,9 +430,18 @@ func (l *Log) WriteBatch(b *Batch) error {
 }
 
 func (l *Log) writeBatch(b *Batch) error {
+	if b.entries[0].index == 0 {
+		return ErrOutOfOrder
+	}
 	// check that all indexes in batch are sane
+	lastIndex := l.lastIndex
+	if lastIndex == 0 {
+		// allow start from arbitrary start index if WAL is empty
+		lastIndex = b.entries[0].index - 1
+	}
 	for i := 0; i < len(b.entries); i++ {
-		if b.entries[i].index != l.lastIndex+uint64(i+1) {
+		if b.entries[i].index != lastIndex+uint64(i+1) {
+			fmt.Println("out of order", b.entries[i].index, lastIndex, i)
```

Comment:
```suggestion
```

Reply:
It seems we can do without these changes